### ISSUE TITLE
Improve and update the docs

### ISSUE THREAD

**Issue by juandavidkincaid** [i18next/i18next-http-middleware#34]
## 🐛 Bug Report

The documentation, is not updated and does not reflect the actual code.
This applies to the initialization options and to options that are described in the docs but in the code are ignored silently, like  `lookupHeader`,  `lookupHeaderRegex`,  `lookupPath`,  `ignoreCase`, `cookiePath`, `cookieSecure`, `cookieSameSite`.

In summary, updating the docs, and being precise of what the plugin does under the hood, like saying it creates clones of i18n when changing the language instead of changing the language globally on every request. (It happen to me and I had to look into the code to know what was happening to the i18n instance, because of the console log when debugging is on).

## To Reproduce
Basically follow the README.md when initializing the plugin

```typescript
// as docs/README.md said
var middleware = require('i18next-http-middleware')

var lngDetector = new middleware.LanguageDetector()
lngDetector.init(options)
```

## Expected behavior

That the configuration works.

```typescript
// what indeed works
lngDetector.init(null, config.i18n);
```

## Your Environment

- *runtime version*: nodejs
- *i18next version*: 20.1.0
- *os*: Linux
- *any other relevant information*: None

**Comment by adrai** [comment]
The languageDetector options are not ignored, if used in the "normal" preferred way:
![image](https://user-images.githubusercontent.com/1086194/113500606-8b635600-951f-11eb-8e16-b69b44408b26.png)
and these are the defaults: https://github.com/i18next/i18next-http-middleware#detector-options

But feel free to create a PR that improves the readme...

**Comment by juandavidkincaid** [comment]
Okay okay, I'm talking about the other initialization alternatives but, I'll try to do the PR then =]
